src/platform/backends/backend_utils/backend_utils.cpp

```diff
@@ -50,10 +53,28 @@ bool can_reach_gateway(const std::string& ip)
 
 auto virtual_switch_subnet(const QString& bridge_name)
 {
-    auto ip_cmd = QString("ip route show | grep %1 | cut -d ' ' -f1 | cut -d '.' -f1-3").arg(bridge_name);
-    return mp::utils::run_cmd_for_output("bash", {"-c", ip_cmd});
-}
+    // CLI equivalent: ip -4 route show | grep ${BRIDGE_NAME} | cut -d ' ' -f1 | cut -d '.' -f1-3
+    QString subnet;
+
+    const auto output =
+        QString::fromStdString(mp::utils::run_cmd_for_output("ip", {"-4", "route", "show"})).split('\n');
+    for (const auto& line : output)
+    {
+        if (line.contains(bridge_name))
+        {
+            subnet = line.section('.', 0, 2);
+            break;
+        }
+    }
+
+    if (subnet.isNull())
+    {
+        mpl::log(mpl::Level::warning, "daemon",
```

Comment:
We don't really care if the subnet exists already or not, so if you really want to print out an information message here, then I would use `debug` as the log level.